app/controllers/api/v1/users_controller.rb

```diff
@@ -13,6 +13,7 @@ def create
         user = User.new(user_params)
 
         if user.save
+          user.welcome_user
```

Comment:
Question about the magic here. How is this different from the callback 
after_create or after_save.

Does 
`user = User.new(user_params)` only return after the new record is completely done?

Reply:
I submit that these changes result in _less_ magic. It is more explicit at the sites where new users are created and saved, that there is where welcoming gets called.

`user.save` is truthy if  the new user object was successfully saved (committed) to the database and `false` if there were any errors in saving. So the positive branch of `if user.save`  is kind of like what we were thinking of doing with the `after_commit` callback, but the difference is that the act of saving a `User` object is not _inseparably_ tied to also welcoming them. Wherever we create and save a user record, with this we have the _option_ to welcome them. The most concrete places this difference is felt is development and test environments (see more on that below). But along the lines of one of my suggested follow-on work items, separating the creation of a user from the welcoming of a new user lets us reuse the welcoming code in different places, e.g. for both email+password signups and via-social-media sign-ups. In whatever fashion it takes to construct a new type of user, that code can decide if and when the welcoming stuff gets called.

## Dev

From the rails console, you can `> FactoryGirl.create_list(:user, 50)` and generate 50 sample users in your development environment without also trying to send email to them or add them to SendGrid and thus generating a bunch of noise with errors about not being able to do so.

## Test 

The test for the `User` model becomes more simple because we don't have to get clever to figure out how to test that the callback fired at the right time. It's super awkward to try to test that a job gets enqueued when creating a user because you need a user to already be created to set up the expectation that the job was enqueued with that user. Talking about and testing callbacks gets tricky, like time travel.